### PR TITLE
[3.x] Move installation of hostname package into dns_domain setup action, called from install cookbook

### PR DESCRIPTION
**Pull request by eantonin**
### Description of changes
* This is to avoid tests in no_internet subnet to fail
* Moved dns_domain resource in common to be called from both slurm and install packages
* Added dns_domain_setup dependency to init_dns recipe

### Tests
* Kitchen on docker and EC2 (in progress)

### Checklist
- [X] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [X] Check all commits' messages are clear, describing what and why vs how.
- [X] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [X] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-common/resources/dns_domain/dns_domain.rb [hunk 1]
```diff
@@ -16,6 +16,13 @@
 use 'partial/_dns_search_domain_ubuntu'
 use 'partial/_dns_search_domain_redhat'
 
+action :setup do
```
Comment (enrico-usai) [inline cookbooks/aws-parallelcluster-common/resources/dns_domain/dns_domain.rb:19]: let's change the default_action to be `:setup`, in this way is consistent with most of the other resources (and be sure we're calling the righ action in the two places).

File: cookbooks/aws-parallelcluster-common/resources/dns_domain/dns_domain_redhat8.rb [hunk 2]
```diff
@@ -17,6 +17,12 @@
 
 use 'partial/_dns_search_domain_redhat'
 
+action :setup do
```
Comment (enrico-usai) [inline cookbooks/aws-parallelcluster-common/resources/dns_domain/dns_domain_redhat8.rb:20]: same here for teh default action

File: kitchen.resources.yml [hunk 3]
```diff
@@ -172,6 +172,7 @@ suites:
     verifier:
       controls:
         - dns_domain_configured
+        - dns_domain_setup
```
Comment (enrico-usai) [inline kitchen.resources.yml]: here, let's call the :configure explicitly.

Comment (eantonin) [reply kitchen.resources.yml]: not needed in this particular point, but yes, I'll call explicitly the configure where needed after changing the default action

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**